sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/test/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filteringTest/DerivedMetricProjectionsTest.java

```diff
@@ -0,0 +1,225 @@
+            "GET /hiThere", new HashMap<>(), new HashMap<>());
+        DependencyDataColumns dependency = new DependencyDataColumns("test.com", 200000L, true, "GET /hiThere", 200,
+            "HTTP", "https://test.com/hiThere?x=y", new HashMap<>(), new HashMap<>());
+        ExceptionDataColumns exception
+            = new ExceptionDataColumns("Exception Message", "Stack Trace", new HashMap<>(), new HashMap<>());
+        TraceDataColumns trace = new TraceDataColumns("Message", new HashMap<>(), new HashMap<>());
+
+        Map<String, AggregationType> projectionInfo = new HashMap<>();
+        projectionInfo.put("id-for-request", AggregationType.SUM);
+        projectionInfo.put("id-for-dependency", AggregationType.SUM);
+        projectionInfo.put("id-for-exception", AggregationType.SUM);
+        projectionInfo.put("id-for-trace", AggregationType.SUM);
+        DerivedMetricProjections projections = new DerivedMetricProjections(projectionInfo);
+
+        for (int i = 0; i < 2; i++) {
+            projections.calculateProjection(dmiRequest, request);
+        }
+
+        for (int i = 0; i < 3; i++) {
+            projections.calculateProjection(dmiDep, dependency);
+        }
+
+        for (int i = 0; i < 4; i++) {
+            projections.calculateProjection(dmiTrace, trace);
+        }
+
+        projections.calculateProjection(dmiException, exception);
+
+        Map<String, Double> finalValues = projections.fetchFinalDerivedMetricValues();
+        assertEquals(finalValues.get("id-for-request"), 2.0);
+        assertEquals(finalValues.get("id-for-dependency"), 3.0);
+        assertEquals(finalValues.get("id-for-exception"), 1.0);
+        assertEquals(finalValues.get("id-for-trace"), 4.0);
+    }
+
+    @Test
+    void testDurationAvgProjection() {
+        testDurationProjectionWith("request-avg", "dependency-avg", AggregationType.AVG, 400.0);
+    }
+
+    @Test
+    void testDurationMinProjection() {
+        testDurationProjectionWith("request-min", "dependency-min", AggregationType.MIN, 200.0);
+    }
+
+    @Test
+    void testDurationMaxProjection() {
+        testDurationProjectionWith("request-max", "dependency-max", AggregationType.MAX, 600.0);
+    }
+
+    @Test
+    void testCustomDimensionAvgProjection() {
+        testCustomDimProjectionWith("request-avg", AggregationType.AVG, 8.0);
+    }
+
+    @Test
+    void testCustomDimensionMinProjection() {
+        testCustomDimProjectionWith("request-min", AggregationType.MIN, 4.0);
+    }
+
+    @Test
+    void testCustomDimensionMaxProjection() {
+        testCustomDimProjectionWith("request-max", AggregationType.MAX, 15.0);
+    }
+
+    @Test
+    void testCustomDimensionSumProjection() {
+        testCustomDimProjectionWith("request-sum", AggregationType.SUM, 24.0);
+    }
+
+    @Test
+    void testInvalidCustomDimensionProjection() {
+        DerivedMetricInfo dmiRequest = createDerivedMetricInfoWithEmptyFilters("request-avg", "Request",
```

Comment:
`"request-avg"` is used several times in the test. A local variable or a constant may make the test easier to read.